src/styles/map.scss

```diff
@@ -0,0 +1,23 @@
+$marker-diameter: 20px;
+
+.image-container {
+  border: 1px solid #ddd;
+  padding: 1rem;
+  position: relative;
+}
+
+.active-marker {
+  position: absolute;
+  border-radius: 100%;
+  width: $marker-diameter;
+  height: $marker-diameter;
+  background: blue;
+}
+
+.selected-marker {
+  position: absolute;
+  border-radius: 100%;
+  width: $marker-diameter;
+  height: $marker-diameter;
+  background: red;
+}
```

Comment:
Let's compose these styles to avoid redundancy, and to better define what each css rule is specifically used for.

First, let's think of what these names represent: `active-marker` and `selected-marker`. It might be a bit unclear which marker is which by looking at it, so let's think of what these actually are. We are really just talking about:
1) the marker
2) how a selected marker is red

So let's first define how a marker should look:
```scss
// we can just call it `marker` because `active` doesn't tell us anything more about it
.marker {
  position: absolute;
  border-radius: 100%;
  width: $marker-diameter;
  height: $marker-diameter;
  background: blue;
}
```

Now let's make the styles for a `selected` marker. We can tie it back to the `marker` class by making it a _modifier_. The purpose of modifiers is to change existing classes' styles, but they don't hold meaning on their own. 

We'll use the pattern of `block--modifier`, where `block` is the component we're styling, and `modifier` is a different style based on some modified visual state.

In this case, all the `selected` modifier does is changes the color to `red`, so that is all the css we should write:
```scss
.marker--selected {
  background: red;
}
```

This pattern is called the Block Element Modifier pattern, or BEM for short. Read more about it [here](https://css-tricks.com/bem-101/).

Reply:
Since we are using scss, we can take advantage of some additional features that aren't available in standard css. The `&` operator in scss refers to the parent selector. That means we can turn this:
```scss
.marker {
  position: absolute;
  border-radius: 100%;
  width: $marker-diameter;
  height: $marker-diameter;
  background: blue;
}

.marker--selected {
  background: red;
}
```
into this:
```scss
.marker {
  position: absolute;
  border-radius: 100%;
  width: $marker-diameter;
  height: $marker-diameter;
  background: blue;

  &--selected {
    background: red;
  }
}
```
Read more about the `&` operator [here](https://css-tricks.com/the-sass-ampersand/)